packages/dds/tree/src/feature-libraries/default-schema/schemaChecker.ts

```diff
@@ -33,6 +33,11 @@ export function isNodeInSchema(
 	node: MapTree,
 	schemaAndPolicy: SchemaAndPolicy,
 ): SchemaValidationErrors {
+	// If the stored schema is completely empty it means the tree is brand new and we shouldn't validate the data.
+	if (schemaAndPolicy.schema.nodeSchema.size === 0) {
+		return SchemaValidationErrors.NoError;
+	}
```

Comment:
As noted in the description, I'm not 100% confident this is the right solution, so looking for feedback.

Reply:
@Abe27342 Craig pointed out your work on the new API for viewing a tree might be relevant to this change. The overall question is if it's possible for the stored schema to be empty at any point besides when we're initializating a tree for the very first time; so we can decide if using empty stored schema as proxy for first-time-setup is robust enough, and skip validation of the input tree against stored schema in that case. Tagging you in case you have any thoughts already based on what you've been doing.

Reply:
the presence of this bug suggests to me that we're not validating against the right schema in the initialization flow, or otherwise doing something wrong thereabouts... what does the call stack look like when hitting this?

that said, yeah I believe it is in some degenerate cases. We only allow people to increase the set of documents they support over time, but you can declare stored schema repositories in multiple different ways which don't support any trees, so you could initialize the tree with one such way and then "upgrade" back to a schema equivalent to the one we start with (see `storedEmptyFieldSchema`, and an empty array for node schema)

Reply:
I don't think those cases are worth losing a ton of sleep over from a practical standpoint, so it seems like checking in something like this fix then thinking about it further would be a fine course of action.